src/functions-and-arrays.js

```diff
@@ -1,130 +1,246 @@
+   if (arr.length === 0){
+     return null
+   }
+   let longestWord = ''
+   for (let i = 0; i < arr.length; i++){
+     if (arr[i].length > longestWord.length) {
+       longestWord = arr[i]
+     }
+   }
+   return longestWord
+ }
+ 
+ 
+ 
+ // Iteration #3: Calculate the sum
+ const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+ 
+ function sumNumbers(nums) {
+   let sum = 0
+   if (nums.length === 0){
+     return 0
+   }
+   for (let i = 0; i < nums.length; i++){
+     sum = sum + nums[i]
+   }
+    return sum
+ }
+ 
+ 
+ 
+ // Iteration #3.1 Bonus:
+ function sum (sumLength) {
+   let len = 0
+ 
+   for (let i = 0; i < sumLength.length; i++){
+     if (typeof sumLength[i] == 'array' || typeof sumLength[i] == 'object'){
+     return "Unsupported data type sir or ma'am"
+ 
+   } else if (typeof sumLength[i] === 'string') {
+     len = len + sumLength[i].length
+   } else if (typeof sumLength[i] === 'number'){
+     len = len + sumLength[i]
+   } else {
+     len += sumLength[i]
+   }
+  }
+     return len
+ }
+ 
+ 
+ 
+ // Iteration #4: Calculate the average
+ // Level 1: Array of numbers
+ const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+ 
+ function averageNumbers(average) {
+   if (average.length === 0){
+     return null
+     }
+   let total = sumNumbers(average)
+   return total/average.length
+ }
+ 
+ 
+ // Level 2: Array of strings
+ const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+ 
+ function averageWordLength(avgWords) {
+   if (avgWords.length === 0){
+     return null
+     }
+   
+     let totalAvg = 0
+     for (let i = 0; i < avgWords.length; i++){
+       totalAvg = totalAvg + avgWords[i].length
+     }
+  
+   return totalAvg/avgWords.length
+   }
+ 
+ // Bonus - Iteration #4.1
+ function avg(arr) {
+   if (arr.length === 0){
+     return null
+     }
+   
+   let avgSum = 0 + sum(arr)
+   let avgArray = avgSum/arr.length
+   return parseFloat(avgArray.toFixed(2))
+ }
+ 
+ // Iteration #5: Unique arrays
+ const wordsUnique = [
+   'crab',
+   'poison',
+   'contagious',
+   'simple',
+   'bring',
+   'sharp',
+   'playground',
+   'poison',
+   'communion',
+   'simple',
+   'bring'
+ ];
+ 
+ function uniquifyArray(unique) {
+  if (unique.length === 0){
+    return null
+    }  
+  let without = []
+      for (let i = 0; i < unique.length; i++){
+        if (unique.indexOf (unique[i], unique[i]+1) !== unique[i]) {
+           without[i].push
+         }
+       }
+       return without
+ }
+ 
+ 
+ 
+ // Iteration #6: Find elements
+ const wordsFind = ['machine', 'subset', 'trouble', 'starting', 'matter', 'eating', 'truth', 'disobedience'];
+ 
+ function doesWordExist(wFind, word) {
+  if (wFind.length === 0 || word === 0){
+    return null
+    }
+  for (let i = 0; i < wFind.length; i++){
+    if (wFind[i] === word){
+      return true
+    } 
+    }
+  for (let i = 0; i < wFind.length; i++){
+    if (wFind[i] !== word){
+      return false
+    }
```

Comment:
aqui daria pra fazer um `else` no `for` acima, dai evitaria fazer dois loops, o que acha?